### PR TITLE
Fix up instances of E-mail to email

### LINKED ISSUE #22984: E-mail and Email Is Spelled Differently on Page
### MDN URL

https://developer.mozilla.org/en-US/docs/Learn/HTML/Introduction_to_HTML/Creating_hyperlinks

### What specific section or headline is this issue about?

E-mail links

### What information was incorrect, unhelpful, or incomplete?

On this page the heading "E-mail links" is spelled with a dash but in other parts of the page there is no dash. I assume you have a standard way of spelling the word.

I'm not sure how the "In this article" navigation is created so that file might nee to be updated too.

### What did you expect to see?

I expected to see a consistent spelling of the word e-mail whichever way you spell it.

### Do you have any supporting links, references, or citations?

no

### Do you have anything more you want to share?

no

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/learn/html/introduction_to_html/creating_hyperlinks`
* MDN URL: https://developer.mozilla.org/en-US/docs/Learn/HTML/Introduction_to_HTML/Creating_hyperlinks
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/learn/html/introduction_to_html/creating_hyperlinks/index.md
* Last commit: https://github.com/mdn/content/commit/d02c96125b0958bf5bd1bcaf345cb997cf48ce13
* Document last modified: 2022-11-14T23:47:20.000Z

</details>

### PR DESCRIPTION
**Pull request by hamishwillee**
Fixes #22984

"email" is the format mandated by the MDN standard. 

This was done using a case-sensitive word match search-replace, then tiding up the last few ones like "e-mailing" and `_e-mail_` in anchor links.

### CODE REVIEW

**Comment by hamishwillee** [comment]
@nschonni Is this kind of word something we could check for errors in? Following this there is just one "wrong" instance, which is the one in the guide that explains the correct form to use.

File: files/en-us/web/html/element/a/index.md [hunk 1]
```diff
@@ -171,7 +171,7 @@ To create links that open in the user's email program to let them send a new mes
 <a href="mailto:nowhere@mozilla.org">Send email to nowhere</a>
 ```
 
-For details about `mailto:` URLs, such as including a subject or body, see [Email links](/en-US/docs/Learn/HTML/Introduction_to_HTML/Creating_hyperlinks#e-mail_links) or {{RFC(6068)}}.
+For details about `mailto:` URLs, such as including a subject or body, see [Email links](/en-US/docs/Learn/HTML/Introduction_to_HTML/Creating_hyperlinks#email_links) or {{RFC(6068)}}.
```
Comment (nschonni) [inline files/en-us/web/html/element/a/index.md:174]: changing the anchor links in this and a few other places probably isn't correct

**Review by nschonni — COMMENTED** [review]
You could try and enforce this with @OnkarRuikar custom rule as a base https://github.com/mdn/content/blob/b02e17dc1c78083d4b1168711dfbfd7063b6197c/.markdownlint-cli2.jsonc#L122
I guess this lines up with the MS guidance too https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/email

File: files/en-us/web/html/element/a/index.md (hunk 1 above)
Comment (hamishwillee) [reply files/en-us/web/html/element/a/index.md:174]: Thanks @nschonni. It is correct because I fixed up headings as well. I manually checked these.

**Comment by OnkarRuikar** [comment]
> You could try and enforce this with @OnkarRuikar custom rule as a base

This way we can enforce other words too, e.g. `web site` -> `website`.
But doing this in PR checks and husky is not a good idea. As arguments may ensue between PR authors and reviewers.
I think daily lint cron is the better place for such nitpicks. In the Github workflow we can inject such rules in the config before linting. Or have separate sub-config file which will be imported/merged into the main config in the cron job.

**Review by wbamberg — APPROVED** [review]
👍 thank you @hamishwillee !

**Comment by wbamberg** [comment]
> > You could try and enforce this with @OnkarRuikar custom rule as a base
> 
> This way we can enforce other words too, e.g. `web site` -> `website`. But doing this in PR checks and husky is not a good idea. 

Not sure I agree with this. If we have written guidelines for spellings we check, then there should not be an argument (it does help a lot to have a written policy we can point to, so it feels less arbitrary). I've reviewed thousands of MDN PRs at this point, and I can't think of any cases where people have argued with a spelling revision.

**Comment by OnkarRuikar** [comment]
> If we have written guidelines for spellings we check

I think we need a separate word list page like [other doc projects](https://developers.google.com/style/word-list). Because there are going to be more words to correct. We can't keep adding to the example list.

**Pull request merged**